lib/devcycle-ruby-server-sdk/localbucketing/config_manager.rb

```diff
@@ -53,22 +55,43 @@ def fetch_config
           Accept: "application/json",
         })
 
+      begin
+        Date.parse(@config_last_modified)
+        if @config_last_modified != ""
+          req.options[:headers]["If-Modified-Since"] = Time.httpdate(@config_last_modified)
+        end
+      rescue
+      end
+
+
       if @config_e_tag != ""
         req.options[:headers]['If-None-Match'] = @config_e_tag
       end
 
       @max_config_retries.times do
-        @logger.debug("Requesting new config from #{get_config_url}, current etag: #{@config_e_tag}")
+        @logger.debug("Requesting new config from #{get_config_url}, current etag: #{@config_e_tag}, last modified: #{@config_last_modified}")
         resp = req.run
         @logger.debug("Config request complete, status: #{resp.code}")
         case resp.code
         when 304
-          @logger.debug("Config not modified, using cache, etag: #{@config_e_tag}")
+          @logger.debug("Config not modified, using cache, etag: #{@config_e_tag}, last modified: #{@config_last_modified}")
           break
         when 200
-          @logger.debug("New config received, etag: #{resp.headers['Etag']}")
-          set_config(resp.body, resp.headers['Etag'])
-          @logger.debug("New config stored, etag: #{@config_e_tag}")
+          @logger.debug("New config received, etag: #{resp.headers['Etag']} LM:#{resp.headers['Last-Modified']}")
+          lm_header = resp.headers['Last-Modified']
+          begin
+            lm_timestamp = Time.rfc2822(lm_header)
+            current_lm = Time.rfc2822(@config_last_modified)
+            if lm_timestamp == "" && @config_last_modified == "" || (current_lm.utc < lm_timestamp.utc)
```

Comment:
are there any null / nils in ruby?

Reply:
https://blog.arkency.com/2017/07/nil-empty-blank-ruby-rails-difference/